pkg/diagnostics/networkpod/util/log.go

```diff
@@ -0,0 +1,156 @@
+	l.Run([]string{"oc", "get", "nodes", "-o", "yaml"}, "/nodes", p)
+	l.Run([]string{"oc", "get", "pods", "--all-namespaces", "-o", "yaml"}, "/pods", p)
+	l.Run([]string{"oc", "get", "services", "--all-namespaces", "-o", "yaml"}, "/services", p)
+	l.Run([]string{"oc", "get", "endpoints", "--all-namespaces", "-o", "yaml"}, "/endpoints", p)
+	l.Run([]string{"oc", "get", "routes", "--all-namespaces", "-o", "yaml"}, "/routes", p)
+	l.Run([]string{"oc", "get", "clusternetwork", "-o", "yaml"}, "/clusternetwork", p)
+	l.Run([]string{"oc", "get", "hostsubnets", "-o", "yaml"}, "/hostsubnets", p)
+	l.Run([]string{"oc", "get", "netnamespaces", "-o", "yaml"}, "/netnamespaces", p)
+}
+
+func (l *LogInterface) LogServices(logdir string) {
+	p := logdir
+
+	type Service struct {
+		name   string
+		suffix string
+		args   string
+	}
+	allServices := []Service{
+		{suffix: "master", args: "master"},
+		{suffix: "master-controllers", args: "master controllers"},
+		{suffix: "api", args: "master api"},
+		{suffix: "node", args: "node"},
+	}
+	foundServices := []Service{}
+
+	for _, sysDir := range []string{"/etc/systemd/system", "/usr/lib/systemd/system"} {
+		for _, prefix := range []string{"openshift", "origin", "atomic-openshift"} {
+			for _, service := range allServices {
+				service.name = fmt.Sprintf("%s-%s", prefix, service.suffix)
+				servicePath := fmt.Sprintf("%s/%s.service", sysDir, service.name)
+				if _, err := os.Stat(servicePath); err == nil {
+					foundServices = append(foundServices, service)
+				}
+			}
+		}
+	}
+
+	for _, service := range foundServices {
+		l.Run([]string{"journalctl", "-u", service.name, "-r", "-n", "5000"}, fmt.Sprintf("/journal-%s", service.name), p)
+		l.Run([]string{"systemctl", "show", service.name}, fmt.Sprintf("/systemctl-show-%s", service.name), p)
+
+		configFile := l.getConfigFileForService(service.name, service.args)
+		if len(configFile) > 0 {
+			l.Run([]string{"cat", configFile}, fmt.Sprintf("/config-%s", service.name), p)
+		}
+	}
+}
+
+func (l *LogInterface) LogSystem(logdir string) {
+	p := logdir
+
+	l.Run([]string{"journalctl", "--boot", "-r", "-n", "5000"}, "/journal-boot", p)
+	l.Run([]string{"nmcli", "--nocheck", "-f", "all", "dev", "show"}, "/nmcli-dev", p)
+	l.Run([]string{"nmcli", "--nocheck", "-f", "all", "con", "show"}, "/nmcli-con", p)
+	l.Run([]string{"head", "-1000", "/etc/sysconfig/network-scripts/ifcfg-*"}, "/ifcfg", p)
+	l.Run([]string{"ip", "addr", "show"}, "/addresses", p)
+	l.Run([]string{"ip", "route", "show"}, "/routes", p)
+	l.Run([]string{"ip", "neighbor", "show"}, "/arp", p)
+	l.Run([]string{"iptables-save"}, "/iptables", p)
+	l.Run([]string{"cat", "/etc/hosts"}, "/hosts", p)
+	l.Run([]string{"cat", "/etc/resolv.conf"}, "/resolv.conf", p)
+	l.Run([]string{"lsmod"}, "/modules", p)
+	l.Run([]string{"sysctl -a"}, "/sysctl", p)
+	l.Run([]string{"oc version"}, "/version", p)
+	l.Run([]string{"docker version"}, "/version", p)
+	l.Run([]string{"cat /etc/system-release-cpe"}, "/version", p)
```

Comment:
In the last few lines, you're passing a single arg with spaces in it rather than multiple args. Although that ends up doing the same thing anyway... which is odd.